lib/zoneddatetime.ts

```diff
@@ -606,8 +607,8 @@ export class ZonedDateTime implements Temporal.ZonedDateTime {
     const dtStart = new DateTime(GetSlot(dt, ISO_YEAR), GetSlot(dt, ISO_MONTH), GetSlot(dt, ISO_DAY), 0, 0, 0, 0, 0, 0);
     const instantStart = ES.BuiltinTimeZoneGetInstantFor(timeZone, dtStart, 'compatible');
     const endNs = ES.AddZonedDateTime(instantStart, timeZone, calendar, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0);
-    const dayLengthNs = endNs.subtract(GetSlot(instantStart, EPOCHNANOSECONDS));
-    if (dayLengthNs.isZero()) {
+    const dayLengthNs = JSBI.subtract(endNs, JSBI.BigInt(GetSlot(instantStart, EPOCHNANOSECONDS)));
+    if (JSBI.equal(dayLengthNs, ZERO)) {
```

Comment:
I was treating zero specially simply because bigInteger has the `isZero` method, and so I started by quickly find/replacing this with a reference to the constant.

I agree we should dedupe the others, but I don't know how best to name the constants. THOUSAND, MILLION, <whatever 1e9 is called> ? Open to suggestions.